src/factories/watchableListFactory.js

```diff
@@ -0,0 +1,72 @@
+angular.module('bicker_router').factory('WatchableListFactory', function(ObjectHelper, WatcherFactory) {
+  let factory;
+  class WatchableList {
+    constructor(list) {
+      this.list = list;
+      this.watchers = [];
+    }
+
+    get(path) {
+      return ObjectHelper.get(this.list, path);
+    }
+
+    getAll() { return this.list; }
+
+    getSubset(paths) {
+      return _.zipObject(paths, _.map(paths, this.get.bind(this)));
+    }
+
+    set(path, value) {
+      ObjectHelper.set(this.list, path, value);
+      return this._notifyWatchers(path, value);
+    }
+
+    unset(paths) {
+      if (!(paths instanceof Array)) { paths = [paths]; }
+
+      return Array.from(paths).map((path) =>
+        (ObjectHelper.unset(this.list, path),
+        this._notifyWatchers(path, undefined)));
+    }
+
+    watch(paths, handler) {
+      if (!(paths instanceof Array)) { paths = [paths]; }
+
+      return Array.from(paths).map((path) =>
+        this.watchers.push(WatcherFactory.create(path, handler, this.get(path))));
+    }
+
+    removeWatcher(watcher) {
+      if (this.watchers.length === 0) { return; }
+      let newWatchers = [];
+
```

Comment:
Looks like whatever you used to convert this really did a number on this range expression... please tidy up this for loop into something a bit more readable